### ISSUE TITLE
./Telegram: symbol lookup error: ./Telegram: undefined symbol: g_resolver_set_timeout

### ISSUE THREAD

**Issue by kokkekpek** [telegramdesktop/tdesktop#26601]
### Steps to reproduce

1. Successfuly build Telegram on Linux from source using [this instruction](https://github.com/telegramdesktop/tdesktop/blob/dev/docs/building-linux.md)
2. Run `./Telegram` from `out/Debug` directory
3. Get this error: `./Telegram: symbol lookup error: ./Telegram: undefined symbol: g_resolver_set_timeout`

# Aditional info

## Shared object dependencies

```shell
ldd Telegram 
```

```txt
	linux-vdso.so.1 (0x00007ffeb0bd7000)
	libgio-2.0.so.0 => not found
	libgobject-2.0.so.0 => /usr/lib64/libgobject-2.0.so.0 (0x00007f58e403a000)
	libfontconfig.so.1 => /usr/lib64/libfontconfig.so.1 (0x00007f58e3fed000)
	libfreetype.so.6 => /usr/lib64/libfreetype.so.6 (0x00007f58e3f20000)
	libEGL.so.1 => /usr/lib64/libEGL.so.1 (0x00007f58e3f0a000)
	libGL.so.1 => /usr/lib64/libGL.so.1 (0x00007f58e3e84000)
	libxcb.so.1 => /usr/lib64/libxcb.so.1 (0x00007f58e3e58000)
	libX11.so.6 => /usr/lib64/libX11.so.6 (0x00007f58e3d16000)
	libX11-xcb.so.1 => /usr/lib64/libX11-xcb.so.1 (0x00007f58e3d11000)
	libglib-2.0.so.0 => /usr/lib64/libglib-2.0.so.0 (0x00007f58e3bce000)
	libdl.so.2 => /usr/lib64/libdl.so.2 (0x00007f58e3bc9000)
	libm.so.6 => /usr/lib64/libm.so.6 (0x00007f58e3aec000)
	libpthread.so.0 => /usr/lib64/libpthread.so.0 (0x00007f58e3ae5000)
	libc.so.6 => /usr/lib64/libc.so.6 (0x00007f58e3911000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f58e40c1000)
	libffi.so.8 => /usr/lib64/libffi.so.8 (0x00007f58e3905000)
	libexpat.so.1 => /usr/lib64/libexpat.so.1 (0x00007f58e38da000)
	libz.so.1 => /usr/lib64/libz.so.1 (0x00007f58e38c0000)
	libbz2.so.1 => /usr/lib64/libbz2.so.1 (0x00007f58e38ad000)
	libpng16.so.16 => /usr/lib64/libpng16.so.16 (0x00007f58e3874000)
	libbrotlidec.so.1 => /usr/lib64/libbrotlidec.so.1 (0x00007f58e3867000)
	libGLdispatch.so.0 => /usr/lib64/libGLdispatch.so.0 (0x00007f58e37ae000)
	libGLX.so.0 => /usr/lib64/libGLX.so.0 (0x00007f58e377a000)
	libXau.so.6 => /usr/lib64/libXau.so.6 (0x00007f58e3775000)
	libXdmcp.so.6 => /usr/lib64/libXdmcp.so.6 (0x00007f58e376d000)
	libpcre2-8.so.0 => /usr/lib64/libpcre2-8.so.0 (0x00007f58e36d0000)
	libbrotlicommon.so.1 => /usr/lib64/libbrotlicommon.so.1 (0x00007f58e36ad000)
```

## SO stats

```shell
stat /usr/lib64/libgio-2.0.so
```

```shell
  File: /usr/lib64/libgio-2.0.so -> libgio-2.0.so.0
  Size: 15        	Blocks: 0          IO Block: 4096   symbolic link
Device: 254,0	Inode: 1055253     Links: 1
Access: (0777/lrwxrwxrwx)  Uid: (    0/    root)   Gid: (    0/    root)
Access: 2023-08-02 00:06:40.367943667 +0300
Modify: 2023-07-03 03:04:02.400070325 +0300
Change: 2023-07-03 03:04:04.406676150 +0300
 Birth: 2023-07-03 03:04:02.400070325 +0300
```

```shell
stat /usr/lib64/libgio-2.0.so.0
```

```shell
  File: /usr/lib64/libgio-2.0.so.0 -> libgio-2.0.so.0.7600.3
  Size: 22        	Blocks: 0          IO Block: 4096   symbolic link
Device: 254,0	Inode: 1055252     Links: 1
Access: (0777/lrwxrwxrwx)  Uid: (    0/    root)   Gid: (    0/    root)
Access: 2023-08-01 23:57:55.578361915 +0300
Modify: 2023-07-03 03:04:02.400070325 +0300
Change: 2023-08-01 23:57:54.518427110 +0300
 Birth: 2023-07-03 03:04:02.400070325 +0300
```

```shell
stat /usr/lib64/libgio-2.0.so.0.7600.3
```

```shell
  File: /usr/lib64/libgio-2.0.so.0.7600.3
  Size: 1972728   	Blocks: 3856       IO Block: 4096   regular file
Device: 254,0	Inode: 1054552     Links: 1
Access: (0755/-rwxr-xr-x)  Uid: (    0/    root)   Gid: (    0/    root)
Access: 2023-08-01 20:43:48.672537504 +0300
Modify: 2023-07-03 03:04:03.423372631 +0300
Change: 2023-07-03 03:04:04.406676150 +0300
 Birth: 2023-07-03 03:04:02.276740730 +0300
```

## Search `g_resolver_set_timeout` in SO

```shell
stat /usr/lib64/libgio-2.0.so.0.7600.3 | grep g_resolver_set_timeout
```

Nothing

### Expected behaviour

Launch Telegram and watch videos with cats

### Actual behaviour

Can't  launch Telegram and watch videos with cats

### Operating system

Gentoo, Kernel: 6.1.41, Qt: 5.15.10, KDE Frameworks: 5.108.0, kf5-config: 1.0

### Version of Telegram Desktop

v4.8.10, commit: 8ec0bdcac935adfdae740c4551db08927be59dd7

### Installation source

Static binary from official website

**Comment by kokkekpek** [comment]
I found that `g_resolver_set_timeout` in `libgio` is only available since version `2.78` - [code](https://github.com/GNOME/glib/blob/main/gio/gresolver.h#L282C4-L282C4)

But for `2.78` version of `libgio` doesn't exist. Last version is `2.77.1` - [code](https://github.com/GNOME/glib/blob/main/NEWS#L1)

And for Gentoo last stable glib version) is `2.76.3` -  [link](https://packages.gentoo.org/packages/dev-libs/glib)

But after Telegram compilation using [this instruction](https://github.com/telegramdesktop/tdesktop/blob/dev/docs/building-linux.md) i found in `out/gen/gio/resolver_impl.hpp` this code:
```shell
// void g_resolver_set_timeout (GResolver* resolver, unsigned timeout_ms);
// void g_resolver_set_timeout (::GResolver* resolver, guint timeout_ms);
void base::ResolverBase::set_timeout (guint timeout_ms) noexcept
{
  typedef void (*call_wrap_t) (::GResolver* resolver, guint timeout_ms);
  call_wrap_t call_wrap_v = (call_wrap_t) g_resolver_set_timeout;
  auto timeout_ms_to_c = timeout_ms;
  call_wrap_v ((::GResolver*) (gobj_()), (guint) (timeout_ms_to_c));
}
```

I think code generation is wrong or `GIO_AVAILABLE_IN_2_78` [here](https://github.com/GNOME/glib/blob/main/gio/gresolver.h#L282C4-L282C4) is bad style

Heed some help to fix it

**Comment by kokkekpek** [comment]
Compilation with `-D CMAKE_INTERPROCEDURAL_OPTIMIZATION=ON` worked 😊 🎉
